libraries/AP_HAL_SITL/sitl_rangefinder.cpp

```diff
@@ -0,0 +1,79 @@
+/*
+  SITL handling
+
+  This simulates a range_finder
+
+  Andrew Tridgell November 2011
+ */
+
+#include <AP_HAL.h>
+#if CONFIG_HAL_BOARD == HAL_BOARD_SITL
+
+#include <AP_HAL_SITL.h>
+#include "AP_HAL_SITL_Namespace.h"
+#include "HAL_SITL_Class.h"
+
+#include <AP_Math.h>
+#include "../AP_RangeFinder/RangeFinder.h"
+#include "../SITL/SITL.h"
+#include <stdio.h>
+
+extern const AP_HAL::HAL& hal;
+
+using namespace HALSITL;
+
+/*
+  setup the range_finder with new input
+  all inputs are in degrees
+ */
+void SITL_State::_update_range_finder(float altitude)
+{
+	if (_range_finder == NULL) {
+        // no range_finder in this sketch
+        return;
+    }
+
+    RangeFinder::RangeFinder_State state;
+    state.instance = 0;
+    state.distance_cm = altitude*100;
+    state.status = RangeFinder::RangeFinder_Good;
+
+    //fprintf(stdout, "2. altitude = %f \n", altitude);
+
+    _range_finder->setHIL(0, state);
+
+//     uint32_t now = hal.scheduler->millis();
+//     // add delay
```

Comment:
I was going to add a delay for range finder model. But I have not decided yet which place is the best for it (Gazebo pligin or ardupilot). I can remove commented code for now and add the delay later.